### PR TITLE
python312Packages.leanblueprint: init at 0.0.10

### PR DESCRIPTION
**Pull request by xhalo32**
## Description of changes

Added `leanblueprint` 0.0.10 to python packages along with missing dependencies:
- `plasTeX` 3.1
- `plastexdepgraph` 0.0.4
- `plastexshowmore` 0.0.2

Added myself to maintainers.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/leanblueprint/default.nix [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    repo = "leanblueprint";
+    owner = "PatrickMassot";
+    rev = "v0.0.10";
+    hash = "sha256-CUYdxEXgTf2vKDiOoeW4RV6tQ6prFhA4qMc0olZtZBM=";
+  };
+
+  nativeBuildInputs = [ setuptools ];
```
Comment (SomeoneSerge) [inline pkgs/development/python-modules/leanblueprint/default.nix:32]:
```suggestion
  build-system = [ setuptools ];
```

File: pkgs/development/python-modules/leanblueprint/default.nix
Comment (SomeoneSerge) [inline pkgs/development/python-modules/leanblueprint/default.nix:1]:
```


       error: evaluation aborted with the following error message: 'lib.customisation.callPackageWith: Function called without required argument "GitPython" at /ofborg/checkout/3/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-0/pkgs/development/python-modules/leanblueprint/default.nix:18, did you mean "gitpython"?'
````

File: pkgs/development/python-modules/leanblueprint/default.nix [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  # dependencies
+  plasTeX,
+  plastexshowmore,
+  plastexdepgraph,
+  click,
+  rich,
+  rich-click,
+  tomlkit,
+  jinja2,
+  GitPython,
```
Comment (SomeoneSerge) [inline pkgs/development/python-modules/leanblueprint/default.nix:18]: I guess it's all lowercase

Comment (xhalo32) [reply pkgs/development/python-modules/leanblueprint/default.nix:18]: Strange, it worked on my machine

**Comment by xhalo32** [comment]
Update: changed `nativeBuildInputs` to `build-system` and fixed `gitpython` as requested from @SomeoneSerge .

**Comment by SomeoneSerge** [comment]
Thank you @xhalo32

**Comment by SomeoneSerge** [comment]
It's an afterthought but it'd also be ok to add `leanblueprint = toPythonApplication python3Packages.leanblueprint` to `all-packages.nix`

**Comment by xhalo32** [comment]
Good point @SomeoneSerge , I'll do that also

**Pull request merged**